packages/transaction-controller/src/utils/feature-flags.ts

```diff
@@ -116,6 +143,61 @@ export function getBatchSizeLimit(
   );
 }
 
+/**
+ * Retrieves the gas estimate fallback for a given chain ID.
+ * Defaults to the default gas estimate fallback if not set.
+ *
+ * @param chainId - The chain ID.
+ * @param messenger - The controller messenger instance.
+ * @returns The gas estimate fallback.
+ */
+export function getGasEstimateFallback(
+  chainId: Hex,
+  messenger: TransactionControllerMessenger,
+): {
+  gasEstimateFallback: number | Hex;
+  isFixedGas: boolean;
+} {
+  const featureFlags = getFeatureFlags(messenger);
+
+  const gasEstimateFallbackPerChain =
+    featureFlags?.[FEATURE_FLAG_TRANSACTIONS]?.gasEstimateFallbacks?.[chainId];
+
+  if (gasEstimateFallbackPerChain) {
+    return {
+      gasEstimateFallback: gasEstimateFallbackPerChain.value,
+      isFixedGas: gasEstimateFallbackPerChain.isFixedGas,
+    };
+  }
+
+  return getDefaultGasEstimateFallback(messenger);
```

Comment:
Could we simplify the logic into a single exported function such as:

```
const gasEstimateFallbackFlags = featureFlags?.[FEATURE_FLAG_TRANSACTIONS]?.gasEstimateFallbacks;

const chainFlags = gasEstimateFallbackFlags?.perChain?.[chainId];

const percentage = chainFlags?.percentage ??
    gasEstimateFallbackFlags?.default?.percentage ??
    DEFAULT_GAS_ESTIMATE_FALLBACK_BLOCK_PERCENT;

const fixed = chainFlags?.fixed ??
    gasEstimateFallbackFlags?.default?.fixed;

return { fixed, percentage };
```

Reply:
Sure, merged those two functions into one.